store/src/java/com/zimbra/cs/extension/ExtensionUtil.java

```diff
@@ -236,6 +236,18 @@ public static synchronized void destroyAll() {
         sInitializedExtensions.clear();
     }
 
+    public static synchronized void destroy(String extName) {
+        ZimbraExtension ext = getExtension(extName);
+        try {
+            RedoableOp.deregisterClassLoader(ext.getClass().getClassLoader());
+            ext.destroy();
```

Comment:
If we're going to make use of this then the following method needs to be updated to keep track of which extension registered the specified path (no change to signature):
https://github.com/Zimbra/zm-mailbox/blob/develop/store/src/java/com/zimbra/cs/extension/ExtensionDispatcherServlet.java#L59

Then the following method needs to be updated to make sure all paths registered by the extension via the above register method are removed (no change to signature):
https://github.com/Zimbra/zm-mailbox/blob/develop/store/src/java/com/zimbra/cs/extension/ExtensionDispatcherServlet.java#L130

This needs to happen to make sure extensions that register multiple paths can unregister them all.

Reply:
Ok let's connect to discuss this.

Reply:
Created separate story for this, since it should be go through QA cycle as well